### ISSUE TITLE
Improve attribute-indentation (or add a new rule) to require proper indenation for HTML

### ISSUE THREAD

**Issue by ventuno** [ember-template-lint/ember-template-lint#446]
The `attribute-indentation` rule currently applies to handlebar elements/components only.
It would be nice to apply it to HTML as well. Not 100% sure if it's better to:
1. Simply extend the existing rule to cover HTML;
1. Keep this existing rule but add a configuration parameter to include/exclude HTML;
1. Create a new rule.

[1] https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/attribute-indentation.md

**Comment by rwjblue** [comment]
I think 2 is probably the best bet. We are preparing for a major version bump soon, so we can change the default config (and enabling this rule in the recommended config) when we do.

**Comment by ventuno** [comment]
Thanks @rwjblue, I created #447 as an early attempt to implement this.
It's just a very rough implementation for void HTML elements (see [1] for the terminology), I'm creating the pull request just to see if I'm headed in the correct direction.
Most of the naming needs to be figured out, I'm just trying to maximize the amount of existing logic that we can reuse.

A few questions:
1. I'm naming the configuration `parse-html-elements` let me know if that sounds reasonable;
1. Currently we position the "closing brackets" (`}}`) in a newline:
```hbs
{{some-component
  param1
}}
```
What should be preferred for HTML?
```html
<!-- option 1 -->
<input
  type="text"
>
```
```html
<!-- option 2 -->
<input
  type="text">
```
I personally prefer option 2, let me know what you think.

[1] https://en.wikipedia.org/wiki/HTML_element#Syntax

**Comment by rwjblue** [comment]
This has actually been on the docket to fix also, see https://github.com/ember-template-lint/ember-template-lint/issues/267. Basically, I want to allow both by default, and have a config option to force one or the other.

**Comment by ventuno** [comment]
Thanks @rwjblue, what if I work on option 1 right now (e.g. existing option) and take up #267 separately right after this one?
I’d like to keep the pull request’s scope as small as possible.

**Comment by rwjblue** [comment]
Sounds like a great plan 😃

**Comment by ventuno** [comment]
Open for review: https://github.com/ember-template-lint/ember-template-lint/pull/447.

**Comment by ventuno** [comment]
Closed by https://github.com/ember-template-lint/ember-template-lint/pull/447.
